### PR TITLE
SunburstView

### PR DESCRIPTION
**Pull request by albert-smit**
simple sunburst view, currently loads dummy data

### CODE REVIEW

File: src/main/java/nl/tudelft/lifetiles/tree/view/SunburstRing.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        double point3X = centerX + outerRadius * Math.sin(angleAlphaNext);
+        double point3Y = centerY - outerRadius * Math.cos(angleAlphaNext);
+
+        double point4X = centerX + innerRadius * Math.sin(angleAlphaNext);
+        double point4Y = centerY - innerRadius * Math.cos(angleAlphaNext);
+
+        // draw the semi-circle
+        MoveTo move1 = new MoveTo(point1X, point1Y);
+        LineTo line12 = new LineTo(point2X, point2Y);
+
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/tree/view/SunburstRing.java]: The code below could use some more comments

File: src/main/java/nl/tudelft/lifetiles/tree/view/SunburstUnit.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import javafx.scene.Group;
+import javafx.scene.shape.Shape;
+import javafx.scene.text.Text;
+
+/**
+ *
+ * @author Albert Smit
+ *
+ */
+public abstract class SunburstUnit extends Group {
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/tree/view/SunburstUnit.java]: "unit" is quite vague. "node" might be better naming

File: src/main/java/nl/tudelft/lifetiles/tree/view/SunburstRing.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        setName(new Text(getValue().getName()));
+        setDisplay(createRing(layer, degreeStart, degreeEnd, centerX, centerY));
+
+        // calculate the positon of the text
+        double radius = CENTER_RADIUS + (layer * RING_WIDTH) + (RING_WIDTH / 2);
+
+        double degreeCenter = degreeStart
+                + SunburstUnit.calculateAngle(degreeStart, degreeEnd) / 2;
+        //convert to radians
+        double angle = degreeCenter * (Math.PI / (CIRCLEDEGREES / 2));
```
Comment (alangerak) [inline src/main/java/nl/tudelft/lifetiles/tree/view/SunburstRing.java]: could also use Math.toRadians(double)

File: src/main/java/nl/tudelft/lifetiles/tree/view/SunburstRing.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        double arcSize = SunburstUnit.calculateAngle(degreeStart, degreeEnd);
+        boolean largeArc = arcSize > (CIRCLEDEGREES / 2);
+
+        // calculate the radii of the two arcs
+        double innerRadius = CENTER_RADIUS + (layer * RING_WIDTH);
+        double outerRadius = innerRadius + RING_WIDTH;
+
+        // convert degrees to radians for Math.sin and Math.cos
+        double angleAlpha = degreeStart * (Math.PI / (CIRCLEDEGREES / 2));
+        double angleAlphaNext = degreeEnd * (Math.PI / (CIRCLEDEGREES / 2));
```
Comment (alangerak) [inline src/main/java/nl/tudelft/lifetiles/tree/view/SunburstRing.java]: Here you can also use Math.toRadians(double)

**Comment by rutgervandenberg** [comment]
- _SunburstRingSegment.java_: createRing() is 70 lines long. Should split that.

File: src/main/java/nl/tudelft/lifetiles/tree/controller/package-info.java [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /**
  * Controller for the tree module.
```
Comment (jorenham) [inline src/main/java/nl/tudelft/lifetiles/tree/controller/package-info.java]: "Controller" should be plural.

**Comment by joswinter** [comment]
Looks good to me. All comments seem to have been fixed.

**Comment by jorenham** [comment]
I agree.

**Comment by alangerak** [comment]
Looks good now.

**Pull request merged**